ox-chameleon.el

```diff
@@ -103,7 +118,133 @@ When set to nil, the current theme will be used.")
+   (ox-chameleon--face-to-css 'highlight-numbers-number ".org-highlight-numbers-number")
+   (ox-chameleon--generate-html-code-style-font-lock)
+   (ox-chameleon--generate-html-rainbow-parens)
+   (ox-chameleon--generate-html-block-names)))
+
+(defun ox-chameleon--generate-html-block-names ()
+  (string-join
+   (mapcar (lambda (lang)
+             (if-let ((symbols (with-temp-buffer
+                                 (org-mode)
+                                 prettify-symbols-alist)))
+                 (let ((begin (alist-get "#+begin_src" symbols nil nil #'string=))
+                       (end (alist-get "#+end_src" symbols nil nil #'string=)))
+                   (format
+                    "pre.src-%s::before { content: '%s %s'; display: block; %s }
+                     pre.src-%s::after  { content: '%s'; display: block; %s }"
+                    (symbol-name (car lang))
+                    begin
+                    (symbol-name (car lang))
+                    (ox-chameleon--face-to-css 'org-block-begin-line)
+                    (symbol-name (car lang))
+                    end
+                    (ox-chameleon--face-to-css 'org-block-end-line)))
+               (format
+                "pre.src-%s::before { content: '%s'; display: block; %s }"
+                (symbol-name (car lang))
+                (string-replace "-" " " (capitalize (symbol-name (car lang))))
+                (ox-chameleon--face-to-css 'org-block-begin-line))))
+           org-babel-load-languages)))
+
+(defun ox-chameleon--generate-html-code-style-font-lock ()
+  (string-join
+   (mapcar (lambda (face)
+             (ox-chameleon--face-to-css
+              face
+              (format ".org-%s" (substring (symbol-name face) 10 -5))))
+           '(font-lock-comment-face
+             font-lock-comment-delimiter-face
+             font-lock-string-face
+             font-lock-doc-face
+             font-lock-doc-markup-face
+             font-lock-keyword-face
+             font-lock-builtin-face
+             font-lock-function-name-face
+             font-lock-variable-name-face
+             font-lock-type-face
+             font-lock-constant-face
+             font-lock-warning-face
+             font-lock-negation-char-face
+             font-lock-preprocessor-face))))
+
+(defun ox-chameleon--face-to-css (face &optional selector)
+  (let ((pre (if selector (format "%s {" selector) ""))
+        (post (if selector "}" "")))
+    (concat pre
+            (string-join
+             (cl-map 'list (lambda (attr)
+                             (let ((val (ox-chameleon--face-attr face (cdr attr) t)))
+                               (when (engrave-faces--check-nondefault (cdr attr) val)
+                                 (format "%s: %s;" (car attr) val))))
+                     '(("color" . :foreground)
+                       ("background" . :background)
+                       ("font-weight" . :weight)
+                       ("font-family" . :family))))
```

Comment:
Stylistically, I feel that the car and the cdr should be switched here. Somehow this feels like putting the cart before the horse.